ui/vite.config.ts

```diff
@@ -0,0 +1,139 @@
+import {defineConfig, normalizePath} from 'vite';
+import {dirname, resolve} from 'path';
+import {fileURLToPath} from 'url';
+import {viteStaticCopy} from 'vite-plugin-static-copy';
+import babel from '@rollup/plugin-babel';
+import inlinesvg from 'postcss-inline-svg';
+import cssimport from 'postcss-import';
+import postcssurl from 'postcss-url';
+
+
+// @ts-ignore
+const __dirname = dirname(fileURLToPath(import.meta.url));
+const cesiumBuild = resolve(__dirname, './node_modules/cesium/Build/Cesium');
+const extensions = ['.ts', '.js'];
+
+export default defineConfig({
+  resolve: {
+    alias: {
+      cesium: normalizePath(resolve(__dirname, 'node_modules/cesium')),
+      './cesium/Build': normalizePath(resolve(__dirname, 'node_modules/cesium/Build')),
+      './cesium': normalizePath(resolve(__dirname, 'node_modules/cesium/Source')),
+      './fomantic-ui-css': normalizePath(resolve(__dirname, 'node_modules/fomantic-ui-css')),
+      './@fontsource/inter': normalizePath(resolve(__dirname, 'node_modules/@fontsource/inter')),
+      'src': normalizePath(resolve(__dirname, 'src')),
+    },
+    extensions,
+  },
+  build: {
+    outDir: 'dist',
+    emptyOutDir: false,
+    minify: 'terser',
+    sourcemap: true,
+    cssCodeSplit: true,
+    rollupOptions: {
+      input: 'index.html',
+      output: {
+        entryFileNames: 'assets/[name]-[hash].js', // Ensure JavaScript output is configured
+        chunkFileNames: 'assets/[name]-[hash].js',
+        assetFileNames: 'assets/[name]-[hash][extname]',
+      },
+      plugins: [
+        babel({
+          babelHelpers: 'bundled',
+          babelrc: false,
+          // this is duplicated in .browserlistrc
+          // https://babeljs.io/docs/en/options#targets
+          targets: 'last 2 Chrome versions, last 2 Firefox versions, last 2 Safari versions, last 2 Edge versions, Edge 18',
+          plugins: [
+            ['@babel/plugin-proposal-decorators', {decoratorsBeforeExport: true, version: '2023-05'}]
+          ],
+          presets: [
+            [
+              '@babel/preset-typescript',
+              {
+                allowDeclareFields: true,
+              }
+            ],
+            [
+              '@babel/preset-env', {
+              //debug: true, // disable to get debug information
+              modules: false,
+
+              useBuiltIns: 'usage', // required to determine list of polyfills according to browserlist
+              corejs: {version: 3, proposals: false},
+            }
+            ]
+          ],
+          // exclude: 'node_modules/**'
+          extensions: extensions,
+          exclude: [
+            'node_modules/cesium/**',
+            'node_modules/core-js/**',
+            'node_modules/@babel/**',
+            'node_modules/**' // yes, this is eXtreme excluding (includes aws-sdk)

```

Comment:
I don't think the first three exclusions are necessary?
```suggestion
            'node_modules/**',

```